flink-table/flink-table-runtime-blink/src/main/java/org/apache/flink/table/dataformat/vector/VectorizedColumnBatch.java

```diff
@@ -132,4 +133,28 @@ public Decimal getDecimal(int rowId, int colId, int precision, int scale) {
 			return Decimal.fromUnscaledBytes(precision, scale, bytes);
 		}
 	}
+
+	public SqlTimestamp getTimestamp(int rowId, int colId, int precision) {
+		if (isNullAt(rowId, colId)) {
+			return null;
+		}
+
+		if (SqlTimestamp.isCompact(precision)) {
```

Comment:
Have you took a look to hive parquet?
How does it save the timestamp?

Reply:
according to https://github.com/apache/parquet-format/blob/master/LogicalTypes.md#timestamp, we support 3 types of Timestamp in PARQUET(MILLIS(int64), MICROS(int64) and NANOS(int96)).